tests/storage/fsp/test_fsp_sr.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+import os
+from lib.common import wait_for, vm_image
+from lib.vdi import VDI
+
+@pytest.mark.usefixtures("fsp_device_config")
+class TestFSPSRCreateDestroy:
+    """
+    Tests that do not use fixtures that setup the SR
+    because they precisely need to test SR creation and destruction.
+    """
+
+    def test_create_and_destroy_sr(self, host, fsp_device_config):
+        pathtodb = fsp_device_config['pathtodb']
+        host.mkdir('', pathtodb + '/links')
+        sr = host.sr_create('fsp', "fsp-local-SR-test", {'file-uri': pathtodb})
+        sr.destroy()
+
+@pytest.mark.usefixtures("fsp_sr", "fsp_device_config")
+class TestFSPVDI:
+    def test_create_and_destroy_VDI(self, host, fsp_sr, fsp_device_config):
+        pathtoshareddir = os.path.split(fsp_device_config['pathtoshareddir'])
+        linkname = 'vdifor' + pathtoshareddir[-1]
```

Comment:
Would this be appropriate?
```suggestion
        linkname = 'vdifor' + os.path.basename(fsp_device_config['pathtoshareddir'])
```

Reply:
I'll drop this after replacing the parameters with a temporary directory.